python/ray/logger.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+import sys
+
+default_logging_level = logging.INFO
+default_logger = logging.getLogger("ray_default")
+default_logger.setLevel(default_logging_level)
+ch = logging.StreamHandler(sys.stderr)
+ch.setLevel(default_logging_level)
+default_logger.addHandler(ch)
```

Comment:
Also, please do not add a default handler. Then in code which uses ray as a module this handler is also added automatically. Code which uses ray should configure logging by itself. Same for logging level in fact.

I think it would be better if you add a handler in `ray start` script, and add an argument to `ray.init` which would (by default, for backwards compatibility) configure logging to some defaults. In fact, you should probably just call `logging.basicConfig()`, which I think would do exactly  the things above: install stream handler and set the default logging level.

Reply:
@mitar It looks like that `logging.basicConfig()` cannot specify the name like `logging.getLogger(__name__)`.

Reply:
I am not sure what you are saying here. `basicConfig` is to configure the root logger to some sane defaults. (The root logger is the one at the top of hierarchy). Like attaching a stream handler and setting some format string and level.

If you want to configure individual logger directly, then you can fetch its instance and configure it on that. Or you use `dictConfig` for a more descriptive way to do it. There are different way to configure stuff. You can even use exiting support for config files. Or even do it remotely over the network. It is pretty crazy what is possible.

Reply:
But in short: producers of logging messages should just do a simple `logger = logging.getLogger(__name__)` and log messages. And do not care about level and handlers and formats. Then, consumers of those messages configure stuff. And Ray can provide some defaults in their entry points (`ray start` and `ray.init`).